seed/static/seed/js/controllers/data_upload_modal_controller.js

```diff
@@ -355,6 +356,47 @@ angular.module('BE.seed.controller.data_upload_modal', [])
         });
       };
 
+      $scope.get_pm_report_template_names = function (pm_username, pm_password) {
+        spinner_utility.show();
+        $scope.pm_buttons_enabled = false;
+        $http.post('/api/v2_1/portfolio_manager/template_list/', {
+          username: pm_username,
+          password: pm_password
+        }).then(function (response) {
+          $scope.pm_templates = response.data.templates;
+          spinner_utility.hide();
+          $scope.pm_buttons_enabled = true;
+          return response.data;
+        });
+      };
+
+      $scope.get_pm_report = function (pm_username, pm_password, pm_template) {
+        spinner_utility.show();
+        $scope.pm_buttons_enabled = false;
+        $http.post('/api/v2_1/portfolio_manager/report/', {
+          username: pm_username,
+          password: pm_password,
+          template: pm_template
+        }).then(function (response) {
+          response = $http.post('/api/v2/upload/create_from_pm_import/', {
+            properties: response.data.properties,
+            import_record_id: $scope.dataset.id
+          });
+          return response;
+        }).then(function (response) {
+          $scope.uploaderfunc(
+            'upload_complete',
+            {
+              file_id: response.data.import_file_id,
+              source_type: 'PortfolioManager',
+              cycle_id: $scope.selectedCycle.id
+            }
+          );
+          spinner_utility.hide();
+          $scope.pm_buttons_enabled = true;
+        });
+      };
+
```

Comment:
I would _love_ @axelstudios to look this over and see if it is OK.  They are small functions, but still, I want to make sure they are being done properly.